src/esp/assets/MeshMetaData.h

```diff
@@ -5,10 +5,46 @@
 #pragma once
 
 #include "esp/core/esp.h"
+#include "esp/gfx/magnum.h"
 
 namespace esp {
 namespace assets {
 
+//! Some mesh files include a transformation hierarchy. A @ref MeshTransformNode
+//! stores this hierarchy and indices for the meshes at each level such that it
+//! can be reused to instances meshes later.
+struct MeshTransformNode {
+  //! mesh ID within @ref MeshMetaData::meshIndex
+  int meshIDLocal;
+
+  //! material ID within @ref MeshMetaData::materialIndex
+  int materialIDLocal;
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform
+  Magnum::Matrix4 T;
```

Comment:
Let's name this more explicitly.  I like the notation of `T_<to>_<from>`, so if this goes matrix that transforms things world to local coordinates, it would be `T_local_world`

I like this notation because you can write `pt_local = T_local_world * pt_world` and it's clear what happens because the `world`'s "cancel" out and you left with just `local`.

This matrix is likely `T_world_local` in this notation.

Reply:
Sure, this sounds reasonable. In this notation `T_parent_local` may be more accurate. 👍